src/NHibernate.Test/NHSpecificTest/NH1632/Fixture.cs

```diff
@@ -84,43 +84,51 @@ public void When_commiting_items_in_DTC_transaction_will_add_items_to_2nd_level_
 			{
 				using (var s = sessions.OpenSession())
 				{
-					s.Save(new Nums {ID = 29, NumA = 1, NumB = 3});
+					s.Save(new Nums { ID = 29, NumA = 1, NumB = 3 });
 				}
 				tx.Complete();
 			}
-
-			using (var tx = new TransactionScope())
+			try
 			{
-				using (var s = sessions.OpenSession())
+
+				using (var tx = new TransactionScope())
 				{
-					var nums = s.Load<Nums>(29);
-					Assert.AreEqual(1, nums.NumA);
-					Assert.AreEqual(3, nums.NumB);
+					using (var s = sessions.OpenSession())
+					{
+						var nums = s.Load<Nums>(29);
+						Assert.AreEqual(1, nums.NumA);
+						Assert.AreEqual(3, nums.NumB);
+					}
+					tx.Complete();
 				}
-				tx.Complete();
-			}
 
-			//closing the connection to ensure we can't really use it.
-			var connection = sessions.ConnectionProvider.GetConnection();
-			sessions.ConnectionProvider.CloseConnection(connection);
+				//closing the connection to ensure we can't really use it.
+				var connection = sessions.ConnectionProvider.GetConnection();
+				sessions.ConnectionProvider.CloseConnection(connection);
+				// The session is supposed to succeed because the second level cache should have the
+				// entity to load, allowing the session to not use the connection at all.
 
-			using (var tx = new TransactionScope())
-			{
-				using (var s = sessions.OpenSession(connection))
+				using (var tx = new TransactionScope())
 				{
-					var nums = s.Load<Nums>(29);
-					Assert.AreEqual(1, nums.NumA);
-					Assert.AreEqual(3, nums.NumB);
+					using (var s = sessions.OpenSession(connection))
+					{
+						Nums nums = null;
+						Assert.DoesNotThrow(() => nums = s.Load<Nums>(29), "Failed loading entity from second level cache.");
```

Comment:
Leaving an explicit message instead of caching failure, instead of failing, then failing to tear down, and letting no much clues.